lang/syn/src/idl/parse/file.rs

```diff
@@ -405,13 +406,35 @@ fn parse_ty_defs(ctx: &CrateContext, no_docs: bool) -> Result<Vec<IdlTypeDefinit
                     IdlEnumVariant { name, fields }
                 })
                 .collect::<Vec<IdlEnumVariant>>();
+
             Some(Ok(IdlTypeDefinition {
                 name,
                 generics: None,
                 docs: doc,
                 ty: IdlTypeDefinitionTy::Enum { variants },
             }))
         }))
+        .chain(ctx.type_aliases().filter_map(|alias| {
+            // Only take public types
+            if !matches!(&alias.vis, syn::Visibility::Public(_)) {
+                return None;
+            }
+
+            let name = alias.ident.to_string();
+            let doc = if !no_docs {
+                docs::parse(&alias.attrs)
+            } else {
+                None
+            };
+            let value = IdlType::from_str(&alias.ty.to_token_stream().to_string()).unwrap();
+
+            Some(Ok(IdlTypeDefinition {
+                name,
+                generics: None,
+                docs: doc,
+                ty: IdlTypeDefinitionTy::Alias { value },
+            }))
```

Comment:
This wouldn't support type aliases with lifetimes right?

Reply:
It supports the types that have `FromStr` implementation. It wouldn't support lifetimes because the IDL implementation itself doesn't support it.